QStore4S/src/main/java/edu/asu/qstore4s/db/neo4j/impl/DbConnector.java

```diff
@@ -850,8 +670,30 @@ private String buildWhereClause(String name, String property, String value,
      */
 
     private String buildWhereCluse(String name, String property, String value) {
-        return "( HAS (" + name + "." + property + ") and " + name + "."
-                + property + "='" + value + "') ";
+        return "( HAS (" + name + "." + property + ") and " + name + "." + property + "='" + value + "') ";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public List<CreationEvent> executeQuery(String query, String clas) {
+        Class<?> classs = null;
+        if ("relationevent".equals(clas.toLowerCase())) {
+            classs = RelationEvent.class;
+        } else {
+            classs = AppellationEvent.class;
+        }
```

Comment:
this is a bad approach. What if we want to return other objects in the future? This method should take a class object, and then you should validate the input string much earlier and turn it into a class object.